src/methods/mergeItems.ts

```diff
@@ -0,0 +1,98 @@
+import log from "../helpers/log.js"
+import { SimpleDataItem, Options, defaultOptions } from "../types.js"
+import showTable from "./showTable.js"
+import SimpleData from "../class/SimpleData.js"
+import checkTypeOfKey from "../helpers/checkTypeOfKey.js"
+import percentage from "../helpers/percentage.js"
+
+export default function mergeItems(data: SimpleDataItem[], dataToBeMerged: SimpleDataItem[], commonKey: string, options: Options): SimpleDataItem[] {
+
+    const start = Date.now()
+
+    options = {
+        ...defaultOptions,
+        ...options
+    }
+
+    options.logs && log("\nmergeItems() " + commonKey)
+    options.logOptions && log("options:")
+    options.logOptions && log(options)
+
+    let newData
+
+    if (dataToBeMerged instanceof SimpleData) {
+        newData = dataToBeMerged.data
+    } else {
+        newData = dataToBeMerged
+    }
+
+    // All items needs to have the same keys
+    if (!newData[0].hasOwnProperty(commonKey)) {
+        throw new Error("No key named " + commonKey + " in data")
+    }
+    if (!newData[0].hasOwnProperty(commonKey)) {
+        throw new Error("No key named " + commonKey + " in dataToBeMerged")
+    }
+
+    const dataKeys = Object.keys(data[0]).filter(d => d !== commonKey)
+    const newDataKeys = Object.keys(newData[0]).filter(d => d !== commonKey)
+
+    for (let key of dataKeys) {
+        for (let newKey of newDataKeys) {
+            if (newKey === key) {
+                throw new Error("Key " + key + " is present in data and in dataToBeMerged. Rename it in data or dataToBeMerged")
+            }
+        }
+    }
+
+    if (!checkTypeOfKey(data, commonKey, "string", 0.5, options)) {
```

Comment:
If you have a big dataset, it may take a lot of time to check all the values. That's why I thought about checking half of them. Maybe I could add a minimum number to check (10,000) and a maximum (let's say 50,000) in the options. And the checkTypeOfKey would use that. What do you think?

Reply:
Similar to a comment I made in this PR #7, should we make these "general" data check in the constructor instead?

Reply:
In the constructor, the check would happen when creating the class. But the user can change the type of the values (from string to dates for example). So we need to recheck depending on the method I think.

Reply:
Hum... I read the code again. You are right. It should be 100%. Correcting it.

Reply:
By default, we are checking the first 1000 values (for performance reasons), but if the data has more items than that and if options.logs is true, it will log a warning.